packages/core/i18n/src/Translation.ts

```diff
@@ -1,10 +1,14 @@
 import { defineComponent, h } from 'vue'
-import type { VNodeChild, App } from 'vue'
+import type { VNodeChild, App, PropType } from 'vue'
 import type { IntlShapeEx } from './types'
 
-const TranslationComponent = (i18n: IntlShapeEx) => defineComponent({
+export const i18nTComponent = (i18n: IntlShapeEx | null = null) => defineComponent({
   name: 'I18nT',
   props: {
+    i18n: {
+      type: Object as PropType<IntlShapeEx>,
+      default: null,
+    },
```

Comment:
question: now that I'm looking at this logic, we always pass in the `i18n` object as a parameter as shown here: 

```ts
const i18nT = i18nTComponent(i18nLocal)
```

So it looks like we never actually pass in the `i18n` prop in any of our documented use-cases because it has to be passed into the function when registered in the component.

**I think this makes sense, because if you used the prop, you would have to pass in the `i18n` object in your `<template>` usage every single time, which would not be ideal.**

So I think we can actually remove the `prop`, and then in the function call above, expect the `i18n: IntlShapeEx` to be a required parameter, meaning we also don't need to fallback to the prop below.

Reply:
in case of you do old `defaultComponent` way you might need to use i18n as prop
```
<template>
  <i18n-t :i18n="i18n"/>
</template>

<script lang="ts">
import {i18nTcomponent, createI18n} from ... 
import english from ... 

defineComponent({
  components: {i18nTcomponent()},
  setup() {
    return {
      i18n: createI18n(English)
    }
})
</script>

Reply:
So then should we document this usage example as well?

Reply:
sure. documented...